r/src/array.cpp

```diff
@@ -325,6 +325,27 @@ inline SEXP simple_Array_to_Vector(const std::shared_ptr<arrow::Array>& array) {
   return vec;
 }
 
+inline SEXP DictionaryArrayInt32Indices_to_Vector(const std::shared_ptr<arrow::Array>& array) {
+  auto p_array = reinterpret_cast<const int*>(
+    array->data()->buffers[1]->data() + array->offset() * sizeof(int));
```

Comment:
can segfault if `buffers[1]` is null. cf https://issues.apache.org/jira/browse/ARROW-3362